### PR TITLE
feat(connector-besu): remove hard dependency on keychain

### LINKED ISSUE #1163: feat(connector-besu): remove hard dependency on keychain

### PR DESCRIPTION
**Pull request by TonyRowntree**
Removed the hard dependency on keychain for connector-besu.

**Added:** 
- getContractInfo method for keychain and JsonObject.
- invokeContractPrivateTx method for private transactions.
- New tests for deploying and invoking the contract using JSON object.

**Updated:** 

- invokeContract method to handle the new getContract methods.
- Tests to use new methods.

Resolves #1163 

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

### CODE REVIEW

**Review by takeutak — APPROVED** [review]

File: packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/invoke-contract-endpoint.ts [hunk 1]
```diff
@@ -87,7 +87,9 @@ export class InvokeContractEndpoint implements IWebServiceEndpoint {
     this.log.debug(reqTag);
     const reqBody: InvokeContractV1Request = req.body;
     try {
-      const resBody = await this.options.connector.invokeContract(reqBody);
+      const resBody = await this.options.connector.getContractInfoKeychain(
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/invoke-contract-endpoint.ts:90]: @TonyRowntree Based on the method name getContractInfoKeychain I would conclude that the invoke contract endpoint does not do any contract invocation. Does it actually do contract invocation? If yes it definitely needs a name that implies that.

File: packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json [hunk 2]
```diff
@@ -816,6 +816,85 @@
...
+                    "contractName",
+                    "contractAbi",
+                    "bytecode",
+                    "web3SigningCredential",
+                    "contractJSON",
+                    "constructorArgs"
+                ],
+                "additionalProperties": false,
+                "properties": {
+                    "contractName": {
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json:831]: @TonyRowntree Is `contractName` being used anywhere in the plugin's method that uses this request object? I couldn't find it.

File: packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json [hunk 3]
```diff
@@ -816,6 +816,85 @@
...
+                        "type": "string",
+                        "description": "The contract name for retrieve the contracts json on the keychain.",
+                        "minLength": 1,
+                        "maxLength": 100,
+                        "nullable": false
+                    },
+                    "contractAbi": {
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json:838]: @TonyRowntree Is `contractAbi` being used anywhere in the plugin's method that uses this request object? I couldn't find it.

**Review by petermetz — CHANGES_REQUESTED** [review]

**Review by petermetz — CHANGES_REQUESTED** [review]
@TonyRowntree The PR description would be a much nicer commit message as well. Right now the commit message is:

```sh
feat(connector-besu): remove hard dependency on keychain 

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

feat(conn): squashclear

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

feat(connector-besu): squashnode

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

feat(connector-besu): squash!!!

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

fix: squashgit rebase upstream/main

Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>
```

**Comment by jagpreetsinghsasan** [comment]
Hi @TonyRowntree 
Any updates on this?

**Comment by petermetz** [comment]
Closing for now as part of the tidying up of the PR backlog. Feel free to re-open anytime!

**Pull request closed without merging**